arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataHandler.java

```diff
@@ -0,0 +1,145 @@
+  }
+
+  @Override
+  public void onBlockAdded(final AddedBlockContext addedBlockContext) {
+    consumeExtraData(addedBlockContext.getBlockHeader());
+  }
+
+  private void consumeExtraData(final BlockHeader blockHeader) {
+    final var rawExtraData = blockHeader.getExtraData();
+
+    if (!Bytes.EMPTY.equals(rawExtraData)) {
+      for (final ExtraDataParser extraDataParser : extraDataParsers) {
+        if (extraDataParser.canParse(rawExtraData)) {
+          final var extraData = rawExtraData.slice(1);
+          extraDataParser.parse(extraData);
+          return;
+        }
+      }
+      log.warn("unsupported extra data field {}", rawExtraData.toHexString());
+    }
+  }
+
+  private interface ExtraDataParser {
+    boolean canParse(Bytes extraData);
+
+    void parse(Bytes extraData);
+
+    static Long toLong(final Bytes fieldBytes) {
+      return UInt32.fromBytes(fieldBytes).toLong();
+    }
+  }
+
+  @SuppressWarnings("rawtypes")
+  private class Version1Parser implements ExtraDataParser {
+    private static final int WEI_IN_KWEI = 1_000;
+    private final LineaProfitabilityConfiguration profitabilityConf;
+    private final FieldConsumer[] fieldsSequence;
+    private final MutableLong currFixedCostKWei = new MutableLong();
+    private final MutableLong currVariableCostKWei = new MutableLong();
+
+    public Version1Parser(final LineaProfitabilityConfiguration profitabilityConf) {
+      this.profitabilityConf = profitabilityConf;
+
+      final FieldConsumer fixedGasCostField =
+          new FieldConsumer<>(
+              "fixedGasCost", 4, ExtraDataParser::toLong, currFixedCostKWei::setValue);
+      final FieldConsumer variableGasCostField =
+          new FieldConsumer<>(
+              "variableGasCost", 4, ExtraDataParser::toLong, currVariableCostKWei::setValue);
+      final FieldConsumer minGasPriceField =
+          new FieldConsumer<>("minGasPrice", 4, ExtraDataParser::toLong, this::updateMinGasPrice);
+
+      this.fieldsSequence =
+          new FieldConsumer[] {fixedGasCostField, variableGasCostField, minGasPriceField};
+    }
+
+    public boolean canParse(final Bytes rawExtraData) {
+      return rawExtraData.get(0) == (byte) 1;
+    }
+
+    public synchronized void parse(final Bytes extraData) {
+      log.info("Parsing extra data version 1: {}", extraData.toHexString());
+      int startIndex = 0;
+      for (final FieldConsumer fieldConsumer : fieldsSequence) {
+        fieldConsumer.accept(extraData.slice(startIndex, fieldConsumer.length));
+        startIndex += fieldConsumer.length;
+      }
+
+      profitabilityConf.updateFixedAndVariableCost(
+          currFixedCostKWei.longValue() * WEI_IN_KWEI,
+          currVariableCostKWei.longValue() * WEI_IN_KWEI);
+    }
+
+    void updateMinGasPrice(final Long minGasPriceKWei) {
+      final var minGasPriceWei = Wei.of(minGasPriceKWei).multiply(WEI_IN_KWEI);
+      final var resp =
+          rpcEndpointService.call(
```

Comment:
What happens if we decide not to expose `miner` api? Will extraData handler also fail?

Reply:
there is a dedicated conf option for in process RPC methods `--Xin-process-rpc-apis` that is reserved for plugin use, and is distinct from the one that exposes RPC methods via HTTP, so there is no need to expose via HTTP if only required by plugins